frontend/services/api.ts

```diff
@@ -19,6 +23,16 @@ const API = axios.create({
   xsrfHeaderName: 'X-CSRF-TOKEN',
 });
 
+const onRequest = (config) => {
+  return {
+    ...config,
+    params: {
+      ...config?.params,
+      locale: getCookie('NEXT_LOCALE') || locales.find((locale) => locale.default).locale,
```

Comment:
We can't really use `useRouter` here, so we're reading the cookie that is set when the user switches languages. If no cookie has been set (for whatever reason), then we default to the default locale set in the config.

Reply:
It is working for client and server sides, so it's perfect.